### PR TITLE
Automakeit: Add all GNU install paths to makeit.

### PR DESCRIPTION
**Pull request by yhcote**
This PR adds the install dir command line options to makeit. This way it is not possible to control where to install different singularity components easily. This also helps with the scripting of RPM spec files and packaging overall.

**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](../CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](../CHANGELOG.md) if necessary according to the [Contribution Guidelines](../CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](../CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](../CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

File: mconfig [hunk 1]
```diff
@@ -35,68 +30,315 @@ tgtstatic=0
...
+    prefix=`echo $1 | sed -e 's|.*\=\(.*\)|\1|'`
+   fi
+   shift;;
+  -e|--exec-prefix*)
+   if [ "$1" = "-e" ]; then
+    if ! echo "$2" | sed -e '/^-.*/q2; /^$/q2' >/dev/null; then
+     echo "error option requires an argument: $1"
+     exit
+    fi
+    prefix=$2
```
Comment (jmstover) [inline mconfig]: Shouldn't this be `exec_prefix`?

File: mlocal/frags/Makefile.stub [hunk 2]
```diff
@@ -183,7 +183,7 @@ $(actions_INSTALL): $(actions_LIST)
 
 .PHONY: man
 man:
-	$(V)go run $(GO_TAGS) $(SOURCEDIR)/src/docs/makeDocs/makeDocs.go $(DESTDIR)$(EXEC_PREFIX)/share/man/man1
+	$(V)go run $(GO_TAGS) $(SOURCEDIR)/src/docs/makeDocs/makeDocs.go $(DESTDIR)$(EXECPREFIX)/share/man/man1
```
Comment (jmstover) [inline mlocal/frags/Makefile.stub]: Now that it exists... this should be:
`$(DESTDIR)$(MANDIR)/man1`

**Review by jmstover — CHANGES_REQUESTED** [review]

File: mlocal/frags/Makefile.stub (hunk 2 above)
Comment (yhcote) [reply mlocal/frags/Makefile.stub]: I just noticed that there a lot wrong in Makefile.stub. Things were put ad hoc.

Consider this PR WIP from now on.

File: mconfig (hunk 1 above)
Comment (yhcote) [reply mconfig]: yes thanks

**Review by jmstover — APPROVED** [review]

**Pull request merged**